Source/qol/autopickup.cpp

```diff
@@ -30,31 +31,73 @@ bool HasRoomForGold()
 	return false;
 }
 
+int NumMiscItemsInInv(int iMiscId)
+{
+	int numItems = 0;
+	for (const auto &item : InventoryAndBeltPlayerItemsRange(Players[MyPlayerId])) {
+		if (item._iMiscId == iMiscId) {
+			numItems++;
+		}
+	}
+	return numItems;

```

Comment:
Add `#include <algorithm>` and:

```suggestion
        InventoryAndBeltPlayerItemsRange items{Players[MyPlayerId]};
	return std::count_if(items.begin(), items.end(), [](const Item &item) { return item._iMiscId == iMiscId; });
```

Reply:
OK that's neat, I wasn't aware of function templates.